staging/src/k8s.io/client-go/util/csaupgrade/upgrade.go

```diff
@@ -0,0 +1,157 @@
+package csaupgrade
+
+import (
+	"bytes"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+// Upgrades the Manager information for fields managed with CSA
+// Prepares fields owned by `csaManager` for 'Update' operations for use now
+// with the given `ssaManager` for `Apply` operations
+//
+// csaManager - Name of FieldManager formerly used for `Update` operations
+// ssaManager - Name of FieldManager formerly used for `Apply` operations
+// subResource - Name of subresource used for api calls or empty string for main resource
```

Comment:
Let's make this doc really solid. A few things to note:
1. This is not reversible,
2. Can be done after server-side has been applied, or before
3. Client-side apply tends to own more fields (including fields that are defaulted), this will possibly remove this defaults, they will be re-defaulted, that's fine.
4. We don't know the context of how this will run yet, but you need to be super careful when you run the update against the object.